specification/applicationinsights/resource-manager/microsoft.insights/stable/2015-05-01/favorites_API.json

```diff
@@ -0,0 +1,403 @@
+          "200": {
+            "description": "The newly created favorite that is associated to the Application Insights component.",
+            "schema": {
+                "$ref": "#/definitions/ApplicationInsightsComponentFavorite"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "FavoriteAdd": {
+            "$ref": "./examples/FavoriteAdd.json"
+          }
+        }
+      },
+      "patch": {
+        "description": "Updates a favorite that has already been added to an Application Insights component.",
+        "operationId": "Favorite_Update",
+        "parameters": [
+          {
+            "$ref": "#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ResourceNameParameter"
+          },
+          {
+            "$ref": "#/parameters/FavoriteIdParameter"
+          },
+          {
+            "name": "favoriteProperties",
+            "description": "Properties that need to be specified to update the existing favorite.",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/ApplicationInsightsComponentFavorite"
+            }
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "The favorite definition updated.",
+            "schema": {
+                "$ref": "#/definitions/ApplicationInsightsComponentFavorite"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "FavoriteList": {
+            "$ref": "./examples/FavoriteUpdate.json"
+          }
+        }
+      },
+      "delete": {
+        "description": "Remove a favorite that is associated to an Application Insights component.",
+        "operationId": "Favorite_Delete",
+        "parameters": [
+          {
+            "$ref": "#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ResourceNameParameter"
+          },
+          {
+            "$ref": "#/parameters/FavoriteIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "The favorite has been successfully removed from the Application Insights component."
+          }
+        },
+        "x-ms-examples": {
+          "FavoriteList": {
+            "$ref": "./examples/FavoriteDelete.json"
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "ApplicationInsightsComponentFavorite": {
+      "description": "Properties that define a favorite that is associated to an Application Insights component.",
+      "properties": {
+        "Name": {
```

Comment:
camel case all property names

Reply:
The service I am defining via Swagger returns “PascalCase” property names and not “camelCase” as defined in our Swagger guidelines.

I can convert the swagger document property names to camelCase but I believe that breaks the auto-generated C# code?

Reply:
Whatever the outcome of the API design discussion, the swagger *must* be true to what the service actually returns. If the service is using PascalCase, then you should add suppressions to tell the linter that you are indeed doing this on purpose. This particular guidelines is about the API design, not about the swagger describing the service :).

Reply:
Thanks for the feedback @johanste, I will add the necessary linter suppressions.

Reply:
Done. I updated suppressions that were already there.